src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -776,6 +796,32 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
         }
         return true;
     }
+    public addGlobalTime(): Promise<void> {
+        const giver: Player = Player.of(this.observerRole);
+        return this.gameService.addGlobalTime(this.currentPartId, this.currentPart, giver);
+    }
+    public addLocalTime(): Promise<void> {
+        const giver: Player = Player.of(this.observerRole);
+        return this.gameService.addLocalTime(giver, this.currentPartId);
+    }
+    public addLocalTimeTo(player: Player, addedMs: number): void {
+        if (player === Player.ZERO) {
+            const currentDuration: number = this.chronoZeroLocal.remainingMs;
+            this.chronoZeroLocal.changeDuration(currentDuration + addedMs);
```

Comment:
quid de faire du branchless programming ici et d'avoir chronos: Map<Player, Chrono> this.chronos[player].changeDuration? Ça évite aussi la branche plus bas, et ça évite de la duplication.